dot/network/notifications.go

```diff
@@ -126,9 +127,10 @@ func (s *Service) createNotificationsMessageHandler(info *notificationsProtocol,
 				info.handshakeData.Store(peer, &handshakeData{
 					validated: false,
 					received:  true,
+					stream:    stream,
 				})
 
-				err := handshakeValidator(peer, hs)
+				err := handshakeValidator(peer, hs, stream)
```

Comment:
instead of passing in the stream to the `handshakeValidator` you can just store it in this function instead